src/dodal/devices/pressure_jump_cell.py

```diff
@@ -60,26 +60,91 @@ class ValveState(StrictEnum):
 
 
 class FastValveState(StrictEnum):
-    FAULT = "Fault"
-    OPEN = "Open"
+    FAULT = ValveState.FAULT.value
+    OPEN = ValveState.OPEN.value
     OPEN_ARMED = "Open Armed"
-    CLOSED = "Closed"
+    CLOSED = ValveState.CLOSED.value
     CLOSED_ARMED = "Closed Armed"
     NONE = "Unused"
 
 
-@dataclass
-class AllValvesControlState:
-    valve_1: ValveControlRequest | None = None
-    valve_3: ValveControlRequest | None = None
-    valve_5: FastValveControlRequest | None = None
-    valve_6: FastValveControlRequest | None = None
+class ValveControlBase(StandardReadable, Movable):
+    open: SignalRW[int]
+    control: SignalRW[ValveControlRequest | FastValveControlRequest]
+
+    def __init__(self, name):
+        if not hasattr(self, "open") or not hasattr(self, "control"):
+            raise TypeError(
+                f"Class {ValveControlBase} requires signals open and control to be defined"
+            )
+
+        super().__init__(name)
+
+    @AsyncStatus.wrap
+    async def _set_open_seq(self):
+        await self.open.set(ValveOpenSeqRequest.OPEN_SEQ.value)
+        await asyncio.sleep(OPENSEQ_PULSE_LENGTH)
+        await self.open.set(ValveOpenSeqRequest.INACTIVE.value)
```

Comment:
If you make the base class an ABC, you can then make use of abstract methods that subclasses have to fill in:

```suggestion
    @property
    @abstractmethod
    def _open(self) -> SignalRW[int]:
        pass

    @AsyncStatus.wrap
    async def _set_open_seq(self):
        await self._open.set(ValveOpenSeqRequest.OPEN_SEQ.value)
        await asyncio.sleep(OPENSEQ_PULSE_LENGTH)
        await self._open.set(ValveOpenSeqRequest.INACTIVE.value)
```

Control doesn't appear to be being used?

Reply:
In fact, looking at how open is being used, I don't think you need to make it this overcomplicated, you can just define a Generic ValveControl that takes either of your enum classes on instantiation and then either opens in this standard way or passes the enum to its control signal.

```python
TValveControlRequest = TypeVar("TValveControlRequest", bound = ValveControlRequest | FastValveControlRequest)

class ValveControl(StandardReadable, Movable[TValveControlRequest], Generic[TValveControlRequest]):

    def __init__(self, prefix: str, valve_control_type: type[TValveControlRequest], name: str = ""):
        with self.add_children_as_readable():
             self.open = epics_signal_rw(int, prefix + ":OPENSEQ")
             self.control = epics_signal_rw(valve_control_type, prefix + ":OPENSEQ")
        super().__init__(name)

    @AsyncStatus.wrap
    async def set(self, value: TValveControlRequest):
        if value == ValveControlRequest.OPEN:
            await self.open.set(ValveOpenSeqRequest.OPEN_SEQ.value)
            await asyncio.sleep(OPENSEQ_PULSE_LENGTH)
            await self.open.set(ValveOpenSeqRequest.INACTIVE.value)
        else:
             await self.control.set(value)


# [...] much further down the page

        self.fast_valve_control = {
            i: ValveControl(f"{prefix}V{i}", FastValveControlRequest) for i in self._fast_valves_numbers
        }

        self.slow_valve_control = {
            i: ValveControl(f"{prefix}V{i}", ValveControlRequest) for i in self._slow_valves_numbers
        }
```

Reply:
I've made a PR of what I think the device could look like here https://github.com/DiamondLightSource/dodal/pull/1386/files

Reply:
Thanks Joseph your suggestion in PR https://github.com/DiamondLightSource/dodal/pull/1386 is much better so will go with that